lib/imagemounter/index.js

```diff
@@ -0,0 +1,119 @@
+        return ret.ImageSignature || [];
+    }
+
+    /**
+     * Check if developer image is mounted.
+     */
+    async isDeveloperImageMounted() {
+        return (await this.lookup()).length > 0;
+    }
+
+    /**
+     * Mount image for device.
+     * @param {string} imageFilePath The file path of image.
+     * @param {string} imageSignatureFilePath The signature file path of given `DeveloperDiskImage.dmg`
+     * @param {string?} imageType Type of image, `Developer` by default.
+     */
+    async mount(imageFilePath, imageSignatureFilePath, imageType = 'Developer') {
+        //check file stats
+        if (!await exists(imageFilePath)) {
+            throw new Error(`The image file provided not exists: ${imageFilePath}`);
+        }
+        const imageFileStat = await lstat(imageFilePath);
+        if (imageFileStat.isDirectory()) {
+            throw new Error(`The given image file is expected to be a file, a directory was given: ${imageFilePath}`);
+        }
+        if (!await exists(imageSignatureFilePath)) {
+            throw new Error(`The signature file provided not exists: ${imageFilePath}`);
+        }
+        if ((await lstat(imageSignatureFilePath)).isDirectory()) {
+            throw new Error(`The given signature file is expected to be a file, a directory was given: ${imageFilePath}`);
+        }
+        //read signature
+        const signature = await new B((resolve, reject) => {
+            readFile(imageSignatureFilePath, (err, data) => {
+                if (err != null) {
+                    reject(err);
+                } else {
+                    resolve(data);
+                }
+            });
+        });
+        const mountedImages = await this.lookup(imageType);
+        if (mountedImages.find((mountedSignature) => signature.equals(mountedSignature))) {
+            log.info(`An image with same signature of ${imageSignatureFilePath} is mounted. Doing nothing here`);
+            return;
+        }
+        //notify device
+        const imageSize = imageFileStat.size;
+        let ret = await this._plistService.sendPlistAndReceive({
+            Command: 'ReceiveBytes',
+            ImageSignature: signature,
+            ImageSize: imageSize,
+            ImageType: imageType
+        });
+        this._checkReturnError(ret);
+        if (ret.Status !== 'ReceiveBytesAck') {
+            throw new Error(`Unexpected return from ${MOBILE_IMAGE_MOUNTER_SERVICE_NAME} on sending ReceiveBytes: ${JSON.stringify(ret)}`);
+        }
+        //push image to device
+        const stream = createReadStream(imageFilePath);
+        try {
+            await new B((resolve, reject) => {
+                stream.on('data', async (data) => {
+                    await this._plistService._socketClient.write(data);
+                });
+                stream.on('end', resolve);
+                stream.on('error', reject);
+            });
+        } finally {
+            stream.close();
+        }
```

Comment:
as written, this will throw an unhandled rejection if writing to the socket fails.

try something like this instead:

```suggestion
        stream.on('data', async (data) => {
          try {
            await this._plistService._socketClient.write(data);
          } catch (err) {
            stream.emit('error', err);
          }
        });
        try {
            await new B((resolve, reject) => {
                stream.on('end', resolve).on('error', reject);
            });
        } finally {
            stream.close();
        }
```